tests/unit/commands/migration/test_migration.py

```diff
@@ -71,6 +75,112 @@ def test_do_list_return_no_migration_message_when_there_is_no_migration(self, mo
         mock_message.assert_called_with("No migrations available")
         mock_migration_table(ANY).assert_not_called()
 
+    @patch("hammr.commands.migration.Migration.download_migration_binary")
+    @patch("hammr.utils.migration_utils.check_migration_configuration")
+    @patch("hammr.utils.migration_utils.check_target")
+    @patch("hammr.utils.migration_utils.check_publish_image")
+    @patch("hammr.utils.migration_utils.check_credAccount")
+    @patch("hammr.commands.migration.Migration.createMigration")
+    @patch("uforge.application.Api._Users._Migrations.Create")
+    @patch("hammr.commands.migration.Migration.deploy_and_launch_agent")
+    @patch("ussclicore.utils.printer.out")
+    def test_do_launch_launch_download_migration_binary_when_all_parameters_are_ok(self, mock_out,
+                                                                                   mock_deploy_and_launch_agent,
+                                                                                   mock_api_create,
+                                                                                   mock_create_Migration,
+                                                                                   mock_check_credAccount,
+                                                                                   mock_check_publish_image,
+                                                                                   mock_check_target,
+                                                                                   mock_check_migration_configuration,
+                                                                                   mock_download):
+        # given
+        mock_check_migration_configuration.return_value = ""
+        mock_check_target.return_value = self.create_target()
+        mock_check_publish_image.return_value = uforge.PublishImageVSphere()
+        mock_check_credAccount.return_value = uforge.CredAccountVSphere()
+        mock_create_Migration.return_value = uforge.migration()
+        m = migration.Migration()
+        m.api = Api("url", username="username", password="password", headers=None,
+                    disable_ssl_certificate_validation=False, timeout=constants.HTTP_TIMEOUT)
+        m.login = "login"
+        m.password = "password"
+
+        mock_deploy_and_launch_agent.return_value(1)
+
+        # when
+        m.do_launch("--file config.json")
+
+        # then
+        mock_out.assert_called_with("file: config.json is validated", "OK")
+        self.assertEquals(mock_download.call_count, 1)
+
+    @patch("hammr.commands.migration.Migration.download_migration_binary")
+    @patch("ussclicore.utils.printer.out")
+    def test_do_launch_display_message_error_when_file_is_not_filled(self, mock_out, mock_download):
```

Comment:
I'm not sure this test is relevant, this tests the ussclicore third party library